From a1e3ef04e1b2970a40111566b149c86838b1f2b9 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Mon, 4 Jan 2021 20:12:40 +0000
Subject: [PATCH] Autoconf 2.70 compat: AM_GNU_GETTEXT

Support for detecting libintl/libiconv, used by gettext(), was added
in commit e8864b1b.  This was an initial step towards being able
to provide language translations.  While the OpenZFS project hasn't
yet added any translations this did make it possible for end users
to add their own.

When this support was added using the AM_GNU_GETTEXT_VERSION macro
was optional.  It was therefore omitted and would be revisited if
OpenZFS ever began providing translations.  As of autoconf 2.70 it is
now mandatory.  Unfortunately, simply adding the missing macro call
isn't desirable since it will result both adding a new dependency on
the autopoint command, and additional macros and directories for
full translation support to the repository.

As a compromise to avoid the autoconf 2.70 error this change renames
the AM_GNU_GETTEXT macro.  This effectively let's us avoid the error
and preserve the current behavior until we either add full support
for translations or remove gettext() support entirely.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
---
 config/gettext.m4      | 8 ++++----
 config/user-gettext.m4 | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/config/gettext.m4 b/config/gettext.m4
index e7832418ea1..adbf5b3c011 100644
--- a/config/gettext.m4
+++ b/config/gettext.m4
@@ -52,17 +52,17 @@ dnl GNU format catalogs when building on a platform with an X/Open gettext),
 dnl but we keep it in order not to force irrelevant filename changes on the
 dnl maintainers.
 dnl
-AC_DEFUN([AM_GNU_GETTEXT],
+AC_DEFUN([CUSTOM_AM_GNU_GETTEXT],
 [
   dnl Argument checking.
   ifelse([$1], [], , [ifelse([$1], [external], , [ifelse([$1], [use-libtool], ,
-    [errprint([ERROR: invalid first argument to AM_GNU_GETTEXT
+    [errprint([ERROR: invalid first argument to CUSTOM_AM_GNU_GETTEXT
 ])])])])
   ifelse(ifelse([$1], [], [old])[]ifelse([$1], [no-libtool], [old]), [old],
-    [errprint([ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported.
+    [errprint([ERROR: Use of CUSTOM_AM_GNU_GETTEXT without [external] argument is no longer supported.
 ])])
   ifelse([$2], [], , [ifelse([$2], [need-ngettext], , [ifelse([$2], [need-formatstring-macros], ,
-    [errprint([ERROR: invalid second argument to AM_GNU_GETTEXT
+    [errprint([ERROR: invalid second argument to CUSTOM_AM_GNU_GETTEXT
 ])])])])
   define([gt_included_intl],
     ifelse([$1], [external], [no], [yes]))
diff --git a/config/user-gettext.m4 b/config/user-gettext.m4
index 824318eab96..b3ac40c1a96 100644
--- a/config/user-gettext.m4
+++ b/config/user-gettext.m4
@@ -2,5 +2,5 @@ dnl #
 dnl # Check if libintl and possibly libiconv are needed for gettext() functionality
 dnl #
 AC_DEFUN([ZFS_AC_CONFIG_USER_GETTEXT], [
-    AM_GNU_GETTEXT([external])
+	CUSTOM_AM_GNU_GETTEXT([external])
 ])
